File: Java/Newton's square root algorithm.java
class GFG 
{ 
	
	
	static double squareRoot(double n, double l) 
	{ 
	
		double x = n; 
	 
		double root; 
	
		
		int count = 0; 
	
		while (true) 
		{ 
			count++; 
	
		
			root = 0.5 * (x + (n / x)); 
	
			
			if (Math.abs(root - x) < l) 
				break; 
	
			x = root; 
		} 
	
		return root; 
	} 
	

	public static void main (String[] args) 
	{ 
		double n = 327; 
		double l = 0.00001; 
	
		System.out.println(squareRoot(n, l)); 
	} 
} 

File: Python3/Newton'ssquarerootalgorithm.py
def squareRoot(n, l) : 

	
	x = n 


	count = 0

	while (1) : 
		count += 1

		
		root = 0.5 * (x + (n / x)) 

		
		if (abs(root - x) < l) : 
			break

	
		x = root 

	return root 


if __name__ == "__main__" : 

	n = 327
	l = 0.00001

	print(squareRoot(n, l)) 

